orc8r/tools/fab/dev_utils.py

```diff
@@ -102,25 +102,31 @@ def construct_magmad_gateway_payload(
 
 def get_next_available_gateway_id(
         network_id: str,
-        admin_cert: types.ClientCert = types.ClientCert(
-            cert='./../../.cache/test_certs/admin_operator.pem',
-            key='./../../.cache/test_certs/admin_operator.key.pem',
-        ),
+        url: str = PORTAL_URL,
+        admin_cert: Optional[types.ClientCert] = None,
```

Comment:
The reason for providing implicit defaults is this way we can pass `None` in as an arg and have it be overridden -- makes it easier for intermediate functions to blindly pass in the admin_cert they received, rather than checking whether it's None

The rc of the linked issue is: the default value is assigned once, rather than on each function call. So if the default value is mutable, it will exhibit unexpected behavior https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

Reply:
You can just use `get_next_available_gateway_id("network_iid")` and it will take the defaults stated on the signature. 

If this were a magma service then that would be different. But this is a fab command/library.  The idea is to be clear on the signatures so when we use it we do not have to know the insights of it. Hiding defaults in the code defeats that purpose. Extensiveness is a trade off for ease and clarity on fab.

On the example  [here](https://sourcegraph.com/-/editor?remote_url=git%40github.com%3Amagma%2Fmagma.git&branch=master&file=feg%2Fgateway%2Ffabfile.py&editor=JetBrains&version=v1.2.1&utm_product_name=IntelliJ+IDEA&utm_product_version=2021.2.2&start_row=243&start_col=17&end_row=243&end_col=17) shows how we ended up with a NONE becoming [1234]. That caused us lots of issues trying to find out why attaches were dropped on our local federated setup.  Should they have used the default on the signature, we would have seen that it immediately when building  the configuration [here](https://sourcegraph.com/-/editor?remote_url=git%40github.com%3Amagma%2Fmagma.git&branch=master&file=feg%2Fgateway%2Ffabfile.py&editor=JetBrains&version=v1.2.1&utm_product_name=IntelliJ+IDEA&utm_product_version=2021.2.2&start_row=390&start_col=16&end_row=390&end_col=16)  (well in that case i would not have used any default at all :D )